README.md

```diff
@@ -420,15 +470,55 @@ is roughly equivalent to the following:
 ```js
 class C {
   constructor() {
-    f.call(this, f, b, "a");
+    f.call(this, "a", b);
   }
 }
 ```
 
-Details:
-- `@initialize` can only be used on public field declarations (including symbols/computed property names), not private.
-- `@initialize` can be used with both static and instance fields.
-- The return value of the callback must be undefined.
+When invoked on something which is not a public field, or when used on the left of another `@initialize` decorator on the same public field, the callback is called without the final "value" argument. The other "property key" argument is also omitted when not available. So this becomes simply a way to schedule work. For example:
+
+```js
+@initialize(f)
+class C { }
+```
+
+is roughly equivalent to the following:
+
+```js
+class C {
+  constructor() {
+    f.call(this);
+  }
+}
+```
+
+The return value is checked to be `undefined`.
+
+### `@expose`
+
+The `@expose` decorator is used on a private class element to expose access to get and set it. It's basically like `@register`, except that callbacks are passed into the provided function to access the element. For example:
+
+```js
+class C {
+  @expose(f) #x;
+}
+```
+
+would behave as:
+
+```js
+class C {
+  @register(proto => f(proto,
+                       "#x",
+                       function() { return this.#x },
+                       function(value) { this.#x = value }))
```

Comment:
The fact that this example references `this` mean that this decorator is run every time a new instance is created?
Have you considered only running `@exposed` once (at declaration time) and passing `receiver` as the first parameter to those accessor functions, to avoid recreating them every time?
Alternatively, if it is run for every instance, I think that the first parameter should be `this` and not `proto`, since private names don't exist on the prototype.

Reply:
It's not very clear what is `this` with each built-in decorators. Only with `@initialize(f)` is defined as class instance.

Reply:
It only runs once, at declaration time. The this reference is because the function is designed to be used with .call. Maybe it should take the receiver as an argument instead, if this is confusing.

Reply:
It only runs once, at declaration time. The this reference is because the function is designed to be used with .call. Maybe it should take the receiver as an argument instead, if this is confusing.

Reply:
Since it will always need to be used with `.call` (otherwise, it would throw), I think that it would be easier if the receiver was the first parameter.

Reply:
OK, I changed both `@initialize` and `@expose` to take the instance as a parameter, for clarity.